annotator-core/src/main/java/edu/ucr/cs/riple/core/metadata/trackers/generatedcode/GeneratedRegionTracker.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 Nima Karimipour
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.core.metadata.trackers.generatedcode;
+
+import edu.ucr.cs.riple.core.metadata.trackers.Region;
+import java.util.Set;
+
+/**
+ * Interface for Trackers which can extend the set of potentially impacted regions with generated
+ * regions by a processor that do not exist in source code. (e.g. <a
+ * href="https://projectlombok.org">Lombok</a>)
+ */
+public interface GeneratedRegionTracker {
+
+  /**
+   * Extents set of received potentially impacted regions with generated regions that can be
+   * impacted as well.
+   *
+   * @param regions Set of potentially impacted regions.
+   * @return Set of potentially impacted regions including all knowing potentially impacted
+   *     generated regions by this tracker.
+   */
```

Comment:
I think this might work better (and we don't need to redundantly do `@param` and `@return`):
```suggestion
   * Given an input set of impacted regions, returns an extended set of potentially-impacted regions, accounting for the properties of impacted regions in code generated by an annotation processor.
   */
```
Needs to be formatted